### PR TITLE
[docs] on multi-asset concept page, use AssetSpec, not AssetOut

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: docs/content/concepts/assets/multi-assets.mdx [hunk 1]
```diff
@@ -28,109 +28,92 @@ The function responsible for computing the contents of any asset is an [op](/con
...
-def downstream1(asset1):
-    # will not run when assets_1_and_2 doesn't materialize the asset1
-    return asset1 + [5]
+@asset(deps=["asset1"])
+def downstream1():
+    """Will not run when assets_1_and_2 doesn't materialize asset1."""
 
 
 @asset
 def downstream2(asset2):
```
Comment (jamiedemaria) [inline docs/content/concepts/assets/multi-assets.mdx:72]: also use  `deps` for  `downstream_2`?

**Review by jamiedemaria — APPROVED** [review]
overall looks good to me! 

one thing i noticed is that when i read the "conditional materialization" and "subsettting" section right after each other i had a moment of confusion about the difference between the two and why `can_subset` isn't required in the conditional materialization case. I think part of it is that these two paragraphs are pretty dense https://github.com/dagster-io/dagster/pull/23111/files#diff-212b6a0eff6255e40aa72e669a5843113f3f2512e889ddd6e08749231ad34d6bR78-R80 which makes it hard to understand when you would use each strategy. Not required for this PR, but it would be good to revisit this section in the future. maybe an example of when you would use each one would be helpful

File: docs/content/concepts/assets/multi-assets.mdx [hunk 2]
```diff
@@ -28,109 +28,92 @@ The function responsible for computing the contents of any asset is an [op](/con
...
-
 ### Conditional materialization
 
-In some cases, an asset may not need to be updated in storage each time the decorated function is executed. You can use the `is_required` parameter along with `yield` syntax to implement this behavior.
+In some cases, an asset may not need to be updated in storage each time the decorated function is executed. You can use the `skippable` parameter along with `yield` syntax and <PyObject object="MaterializeResult" /> to implement this behavior.
 
-If the `is_required` parameter is set to `False` on an output, and your function does not `yield` an <PyObject object="Output" /> object for that output, then no asset materialization event will be created, the I/O manager will not be invoked, downstream assets will not be materialized, and asset sensors monitoring the asset will not trigger.
+If the `skippable` parameter is set to `True` on an <PyObject object="AssetSpec" />, and your function does not `yield` a <PyObject object="MaterializeResult" /> object for that asset, then no asset materialization event will be created, downstream assets in the same run will not be materialized, and asset sensors monitoring the asset will not trigger.
```
Comment (erinkcochran87) [inline docs/content/concepts/assets/multi-assets.mdx:47]:
```suggestion
If the `skippable` parameter is set to `True` on an <PyObject object="AssetSpec" /> and your function does not `yield` a <PyObject object="MaterializeResult" /> object for that asset, then:

- No asset materialization event will be created, 
- Downstream assets in the same run will not be materialized, and 
- Asset sensors monitoring the asset will not trigger
```

I saw Jamie's comment about this paragraph being kinda dense. What d'you think about taking the opportunity to do something like this?

File: docs/content/concepts/assets/multi-assets.mdx [hunk 3]
```diff
@@ -28,109 +28,92 @@ The function responsible for computing the contents of any asset is an [op](/con
...
-    return asset2 + [10]
+    """Will not run when assets_1_and_2 doesn't materialize asset2."""
 ```
 
 ### Subsetting multi-assets
 
 By default, it is assumed that the computation inside of a multi-asset will always produce the contents all of the associated assets. This means that attempting to execute a set of assets that produces some, but not all, of the assets defined by a given multi-asset will result in an error.
 
-Sometimes, the underlying computation is sufficiently flexible to allow for computing an arbitrary subset of the assets associated with it. In these cases, set the `is_required` attribute of the outputs to `False`, and set the `can_subset` parameter of the decorator to `True`.
+Sometimes, the underlying computation is sufficiently flexible to allow for computing an arbitrary subset of the assets associated with it. In these cases, set the `skippable` attribute of the asset specs to `True`, and set the `can_subset` parameter of the decorator to `True`.
```
Comment (erinkcochran87) [inline docs/content/concepts/assets/multi-assets.mdx:80]:
```suggestion
Sometimes, the underlying computation is sufficiently flexible to allow for computing an arbitrary subset of the assets associated with it. In these cases, you should set the following to `True`:

- The asset spec's `skippable` attribute
- The decorator's `can_subset` parameter
```

Similar suggestion here to make this less dense. I'm not sure I formatted 'spec' (or 'specs'?) here correctly, but overall, wdyt of this approach?

File: docs/content/concepts/assets/multi-assets.mdx (hunk 2 above)
Comment (sryza) [reply docs/content/concepts/assets/multi-assets.mdx:47]: 👍

File: docs/content/concepts/assets/multi-assets.mdx (hunk 1 above)
Comment (sryza) [reply docs/content/concepts/assets/multi-assets.mdx:72]: 👍

File: docs/content/concepts/assets/multi-assets.mdx (hunk 3 above)
Comment (sryza) [reply docs/content/concepts/assets/multi-assets.mdx:84]: Mind if I punt this suggestion to a separate change? Seems like a reasonable change to make, but out of scope of what I'm trying to fix in this PR.

**Comment by sryza** [comment]
Going to merge this because I got the approve from Jamie, and the changes are primarily to code snippets. @erinkcochran87 - let me know if there are any followups to the text that you think are important for me to address.

**Pull request merged**